x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtAuthenticationTokenTests.java

```diff
@@ -22,31 +22,42 @@
 
     public void testJwtAuthenticationTokenParse() throws Exception {
         final String signatureAlgorithm = randomFrom(JwtRealmSettings.SUPPORTED_SIGNATURE_ALGORITHMS);
-        final Object secretOrSecretKeyOrKeyPair = JwtTestCase.randomSecretOrSecretKeyOrKeyPair(signatureAlgorithm);
-        final Tuple<JWSSigner, JWSVerifier> jwsSignerAndVerifier = JwtUtil.createJwsSignerJwsVerifier(secretOrSecretKeyOrKeyPair);
-        final String serializedJWTOriginal = JwtTestCase.randomValidSignedJWT(jwsSignerAndVerifier.v1(), signatureAlgorithm).serialize();
+        final JWK jwk = JwtTestCase.randomJwk(JWSAlgorithm.parse(signatureAlgorithm));
```

Comment:
I think we will need a few more test methods in this class:

- Verify that JWTs with invalid signatures are rejected
- Verify JWTs that are tampered are rejected
- Verify unsigned JWTs are rejected
- Verify JWTS with algorithm mixup attack are rejected ( Check OpenIDCOnnectAuthenticatorTests)
- Verify JWTS with None algorithm are rejected
- Verify JWTs with timing issues 

I know you handle some (like timing) in the unit testsof JWTUtil but I'd be keen on having tests that show that authentication fails

Reply:
I agree these tests are good to add. It will add a lot of time to cover these tests.

Reply:
I'm fine to do in a follow-up as long as we mark it with a TODO here and/or a :octocat: issue